src/docfx/restore/PackageResolver.cs

```diff
@@ -152,6 +156,8 @@ private void DownloadGitRepositoryCore(string cwd, string url, string committish
                 }
             }
 
+            GitUtility.Clean(cwd, "-xdff");
```

Comment:
in which case will there be untracked files?

Reply:
I originally thought if something bad happens during checkout, `git clean` may help us. But most likely git index file will also be corrupted by then so this isn't likely to help.

This PR still cannot 100% handle git crash, but would mitigate most of the common scenarios. That is good enough for now.